pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -175,6 +175,7 @@ func (r *Reconciler) reconcile(ctx context.Context, source *v1alpha1.ApiServerSo
 		return err
 	}
 	source.Status.MarkEventTypes()
+	source.Status.ObservedGeneration = source.Generation
```

Comment:
Should this be at the beginning of the reconcile loop? My reasoning being that whatever status this reconcile loop returns, it is associated with `source.Generation`, not whatever the old value of `source.Status.ObservedGeneration` was.

Reply:
Discussed offline. At first, I followed the logic when knative serving update the Status.ObservedGeneration for resources. Talked with devs of Knative serving, they said they will work on update the Status.ObservedGeneration for resources at the beginning of the reconcile loop instead of at the end of the reconcile loop because that's the definition of ObservedGeneration( ObservedGeneration is the 'Generation' of the Service that
 was last processed by the controller). ObservedGeneration is the 'Generation' of the Service that was last processed by the controller no matter the reconcile is successful or not.